src/main/java/com/metamx/http/client/NettyHttpClient.java

```diff
@@ -268,7 +264,12 @@ private void finishRequest()
           public void exceptionCaught(ChannelHandlerContext context, ExceptionEvent event) throws Exception
           {
             if (log.isDebugEnabled()) {
-              log.debug(String.format("[%s] Caught exception", requestDesc), event.getCause());
+              final Throwable cause = event.getCause();
+              if (cause == null) {
```

Comment:
Maybe? unless there's a contract to say it wouldn't I'd rather have a safeguard against future weirdness.